app/src/main/java/com/talmar/conveyor/notificationFilters/BlackListFilter.java

```diff
@@ -0,0 +1,73 @@
+package com.talmar.conveyor.notificationFilters;
+
+import android.app.Notification;
+import android.content.SharedPreferences;
+import android.service.notification.StatusBarNotification;
+
+import androidx.annotation.NonNull;
+
+import com.talmar.conveyor.components.Conversation;
+
+import java.util.regex.Pattern;
+
+
+/**
+ * Filters notifications according to the current echoing mode.
+ */
+public class BlackListFilter implements INotificationFilter, SharedPreferences.OnSharedPreferenceChangeListener {
+    private static final String TITLES_TO_IGNORE_KEY_NAME = "titles_to_ignore";
+
+    private String[] m_titles_to_ignore;
+
+    public BlackListFilter(SharedPreferences sharedPreferences) {
+        m_titles_to_ignore = getTitlesToIgnoreFromPreferences(sharedPreferences);
+
+        sharedPreferences.registerOnSharedPreferenceChangeListener(this);
+    }
+
+    @Override
+    public boolean shouldIgnoreNotification(@NonNull StatusBarNotification sbn) {
+
+        Notification notification = sbn.getNotification();
+        String notificationTemplate = notification.extras.getString(Notification.EXTRA_TEMPLATE);
+        boolean isMessagingStyle = Notification.MessagingStyle.class.getName().equals(notificationTemplate);
+
+        if (!isMessagingStyle) {
+            // This is not a messaging-style notification, meaning that we can't determine if it's
+            // a group or a DM. Ignore it.
+            return true;
+        }
```

Comment:
@SilverTuxedo maybe there should be a filter somewhere at the start that filters out all the non-messaging-style notifications? 

this logic is duplicated